packages/web-components/fast-foundation/src/accordion/accordion.ts

```diff
@@ -57,25 +57,40 @@ export class Accordion extends FoundationElement {
     public accordionItemsChanged(oldValue, newValue): void {
         if (this.$fastController.isConnected) {
             this.removeItemListeners(oldValue);
-            this.accordionIds = this.getItemIds();
             this.setItems();
         }
     }
 
-    private activeid: string;
+    private activeid: string | undefined;
     private activeItemIndex: number = 0;
     private accordionIds: Array<string | null>;
 
     private change = (): void => {
         this.$emit("change");
     };
 
+    private findExpandedItem(): AccordionItem | null {
+        for (let item: number = 0; item < this.accordionItems.length; item++) {
+            if (this.accordionItems[item].getAttribute("expanded") === "true") {
+                return this.accordionItems[item] as AccordionItem;
+            }
+        }
+        return null;
+    }
+
     private setItems = (): void => {
+        if (this.accordionItems.length === 0) {
+            return;
+        }
         this.accordionIds = this.getItemIds();
         this.accordionItems.forEach((item: HTMLElement, index: number) => {
             if (item instanceof AccordionItem) {
                 item.addEventListener("change", this.activeItemChange);
                 if (this.isSingleExpandMode()) {
+                    const expandedItem: AccordionItem | null =
```

Comment:
Do you need to do this work for every item in accordionItems? Seems like it really only needs to happen once since it doesn't use the item or the index from the loop

Reply:
I moved the expandedItem check out of the loop